be/src/exprs/array_functions.cpp

```diff
@@ -1512,4 +1514,149 @@ StatusOr<ColumnPtr> ArrayFunctions::array_intersect_any_type(FunctionContext* ct
+    std::vector<ColumnPtr> non_nullable_key_array_columns(key_array_size);
+
+    for (size_t i = 0; i < key_array_size; ++i) {
+        const ColumnPtr& key_data = key_array_columns[i];
+        if (key_data->is_nullable()) {
+            const auto* key_nullable_column = down_cast<const NullableColumn*>(key_data.get());
+            if (key_nullable_column) {
+                nullable_columns[i] = key_nullable_column;
+                non_nullable_key_array_columns[i] = key_nullable_column->data_column();
+            }
+        } else {
+            non_nullable_key_array_columns[i] = key_data;
+        }
+
+        if (non_nullable_key_array_columns[i] != nullptr) {
+            const auto& key_element_column =
+                    down_cast<ArrayColumn*>(non_nullable_key_array_columns[i].get())->elements_column();
+            const auto& key_offset_column =
+                    down_cast<ArrayColumn*>(non_nullable_key_array_columns[i].get())->offsets_column();
+
+            key_elements_columns[i] = key_element_column;
+            key_offsets_columns[i] = key_offset_column;
+        }
+    }
+
+    SortDescs sort_desc = SortDescs::asc_null_first(src_elements_size);
+    const std::atomic<bool>& cancel = ctx->state()->cancelled_ref();
+    Status status;
+
+    for (size_t i = 0; i < chunk_size; ++i) {
+        if (src_null_map != nullptr && src_null_map->get_data()[i]) {
+            for (auto id = src_offsets_column.get_data()[i]; id < src_offsets_column.get_data()[i + 1]; ++id) {
+                key_sort_index.push_back(static_cast<uint32_t>(id));
+            }
+        } else {
+            auto start_offset = src_offsets_column.get_data()[i];
+            auto end_offset = src_offsets_column.get_data()[i + 1];
+            auto src_column_offset_size = end_offset - start_offset;
+
+            std::vector<ColumnPtr> filtered_key_elements_columns;
+            std::vector<std::shared_ptr<UInt32Column>> filtered_key_offsets_columns;
+
+            for (size_t key_column_size = 0; key_column_size < key_array_size; ++key_column_size) {
+                //  If key_nullable_column->immutable_null_column_data[i] is true, Skip this key column for row i
+                if (nullable_columns[key_column_size] &&
+                    nullable_columns[key_column_size]->immutable_null_column_data()[i]) {
+                    continue;
+                }
+
+                auto key_column_offset_size = key_offsets_columns[key_column_size]->get_data()[i + 1] -
+                                              key_offsets_columns[key_column_size]->get_data()[i];
+                if (src_column_offset_size != key_column_offset_size) {
+                    throw std::runtime_error("Input arrays' size are not equal in array_sortby_multi.");
+                }
+
+                filtered_key_elements_columns.push_back(key_elements_columns[key_column_size]);
+                filtered_key_offsets_columns.push_back(key_offsets_columns[key_column_size]);
```

Comment:
Make `filtered_key_offsets_columns` from `std::vector<std::shared_ptr<UInt32Column>>` to `std::vector<const UInt32Column*>`, and the parameter of `sort_and_tie_columns` also do this change.